src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersRepositoryImpl.java

```diff
@@ -40,14 +42,13 @@ public void deleteAuditEntriesByUuids(List<UUID> uuids) {
 
     @Override
     public void deleteByCaseIds(List<Long> caseIds) {
-
         for (Long cid : caseIds) {
             if (casePaymentOrdersJpaRepository.findAllByCaseId(cid).isEmpty()) {
                 throw new CasePaymentOrderCouldNotBeFoundException(ValidationError.CPO_NOT_FOUND_BY_CASE_ID);
             }
         }
-
         casePaymentOrdersJpaRepository.deleteByCaseIdIsIn(caseIds);
+        casePaymentOrdersJpaRepository.flush();
```

Comment:
may be you can remove flush call now?

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/63/commits/63b750b2aef7a52fd98521c1365f4a7f3c8af243